src/Workspaces/Core/Portable/Formatting/Rules/AbstractFormattingRule.cs

```diff
@@ -6,37 +6,56 @@
+        /// <summary>
+        /// returns AnchorIndentationOperations under a node either by itself or by filtering/replacing operations returned by NextOperation
+        /// </summary>
+        public virtual void AddAnchorIndentationOperations(List<AnchorIndentationOperation> list, SyntaxNode node, OptionSet optionSet, ref NextAction<AnchorIndentationOperation> nextOperation)
         {
             nextOperation.Invoke(list);
         }
 
-        public virtual void AddIndentBlockOperations(List<IndentBlockOperation> list, SyntaxNode node, OptionSet optionSet, NextAction<IndentBlockOperation> nextOperation)
+        /// <summary>
+        /// returns IndentBlockOperations under a node either by itself or by filtering/replacing operations returned by NextOperation
+        /// </summary>
+        public virtual void AddIndentBlockOperations(List<IndentBlockOperation> list, SyntaxNode node, OptionSet optionSet, ref NextAction<IndentBlockOperation> nextOperation)
         {
             nextOperation.Invoke(list);
         }
 
-        public virtual void AddAlignTokensOperations(List<AlignTokensOperation> list, SyntaxNode node, OptionSet optionSet, NextAction<AlignTokensOperation> nextOperation)
+        /// <summary>
+        /// returns AlignTokensOperations under a node either by itself or by filtering/replacing operations returned by NextOperation
+        /// </summary>
+        public virtual void AddAlignTokensOperations(List<AlignTokensOperation> list, SyntaxNode node, OptionSet optionSet, ref NextAction<AlignTokensOperation> nextOperation)
         {
             nextOperation.Invoke(list);
         }
 
-        public virtual AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken previousToken, SyntaxToken currentToken, OptionSet optionSet, NextOperation<AdjustNewLinesOperation> nextOperation)
+        /// <summary>
+        /// returns AdjustNewLinesOperation between two tokens either by itself or by filtering/replacing a operation returned by NextOperation
+        /// </summary>
+        public virtual AdjustNewLinesOperation GetAdjustNewLinesOperation(ref SyntaxToken previousToken, ref SyntaxToken currentToken, OptionSet optionSet, ref NextOperation<AdjustNewLinesOperation> nextOperation)
         {
             return nextOperation.Invoke();
         }
 
-        public virtual AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previousToken, SyntaxToken currentToken, OptionSet optionSet, NextOperation<AdjustSpacesOperation> nextOperation)
+        /// <summary>
+        /// returns AdjustSpacesOperation between two tokens either by itself or by filtering/replacing a operation returned by NextOperation
+        /// </summary>
+        public virtual AdjustSpacesOperation GetAdjustSpacesOperation(ref SyntaxToken previousToken, ref SyntaxToken currentToken, OptionSet optionSet, ref NextOperation<AdjustSpacesOperation> nextOperation)
```

Comment:
These are `ref` instead of `in` because I couldn't find a way to override a method with `in` parameters in VB.

@jaredpar Is there a way to do this?

Reply:
Resolved in this pull request by creating an adapter class `CompatAbstractFormattingRule`...